source/main.cpp

```diff
@@ -98,8 +105,22 @@ int main(int argc, char *argv[])
 		if(!GameData::BeginLoad(argv))
 			return 0;
 		
+		if (! testToRun.empty())
+		{
+			testRunner = new TestRunner((GameData::Tests()).Get(testToRun));
```

Comment:
Where is the call to `delete`? It needs to be called in all cases when `new` is used, including when we are exiting due to an exception.

Reply:
Agreed, I still need to add a delete or solve this in a different way.

Reply:
I fixed this missing delete by removing the TestRunner class completely. The test-status is now tracked through a TestContext class.